rfcs/0013-redesign-assets-id-system.md

```diff
@@ -0,0 +1,255 @@
+and the second for local assets.
+
+While the approach of having two instances of pallet-assets will work for us, 
+it's important to note a difference between the goals of Moonbeam and Picasso.
+
+From Moonbeam's `NormalFilter`:
+> Normal Call Filter
+> We dont allow to create nor mint assets, this for now is disabled
+> We only allow transfers. For now creation of assets will go through
+> asset-manager, while minting/burning only happens through xcm messages
+> This can change in the future
+
+While Moonbeam does not want asset minting or burning to occur outside XCM, 
+on Picasso - minting and burning is a core function of Pablo and Staking.
+
+While using Parity's pallet-assets may be ideal, the lack of lock/hold 
+functionality provided by the interface makes it insufficient for our solution. 
+Therefore, implementing additional traits not found in orml-tokens on our 
+routing layer is instead more ideal, especially since user balances are already 
+on chain.
+
+**Steps**:
+* Import & Declare orml-tokens
+  * We must then declare two instances within our `construct_runtime!` macro.
+
+* Configure Each Instance of orml-tokens
+
+## Update Assets Registry
+
+Assets Registry will be responsible for registering both foreign and local 
+assets. As part of the registration process, asset-meta data should be provided 
+and updatable through assets-registry.
+
+### Pallet Configuration
+
+To be responsible for registering both foreign and local assets, assets-registry
+will need access to the creation functionality of both instances of 
+pallet-assets.
+
+**Steps**:
+
+* Provide configuration item for asset creation to assets-registry
+
+* Provide configuration item for max length of ticker-symbols
+
+### Pallet Storage
+
+To accomplish this, minimal storage modifications will be required. The current 
+pallet storage items do not need to be altered, but several new ones will need 
+to be created.
+
+**Steps**:
+
+* Add a nonce storage item that will be used for generating foreign asset IDs
+
+* Add storage item for Asset ticker-symbol
+
+### Pallet Functions
+
+Assets registry now must support the storage of both local and foreign assets.
+To allow for this, the `register_asset` interface must support both types of 
+assets. 
+
+To enable this, instead of only passing a multi-location, we switch to an enum 
+of either multi-location or local ID.
+
+### Asset ID Creation
+
+To assist in the routing between both instances of pallet-assets, we can 
+dedicate the first 8 bytes to either a pallet ID or other information that 
+determines the source of the asset. The remaining 8 bytes of the asset ID can 
+be the nonce provided by the source.
+
+To create a hash from the nonce, `sp_core::hashing::blake2_64` can be used
+
+**Steps**:
+
+* Provide assets registry with either our previously described local asset ID
+or a multi-location
+
+* If assets registry receives a multi-location, hash the multilocation to create
```

Comment:
IBC location of incoming assets from Composable to Picasso would be `transferPortOnComposable/channelOnComposable/<hash of XcmLocation(1, Here)>` That will be value provided by IBC to `transactor`

Reply:
but then, what asset id will be on Picasso? would it be `<hash of transferPortOnComposable/channelOnComposable/<hash of XcmLocation(1, Here)>` or will try to make XcmLocation(StringIdentifierOfIbcClient, StringIdentifierOfIbcConnection, StringIdentifierOfPort, StringIdentifierOfChannel, ThenPortAgain, ThenChannel, then incoming Hash)`?